lxd/container_lxc.go

```diff
@@ -6777,6 +6777,18 @@ func (c *containerLXC) setNetworkLimits(name string, m types.Device) error {
 		return fmt.Errorf("Network limits are only supported on bridged and p2p interfaces")
 	}
 
+	needsUpdate := false
+	for _, v := range containerNetworkLimitKeys {
+		_, needsUpdate = m[v]
+		if needsUpdate {
+			break
+		}
+	}
+
+	if !needsUpdate {
+		return nil
+	}
```

Comment:
I think this will prevent us from unsetting limits. If I have limits in place and I unset all limits.* keys, this will cause the function to return without ever removing the existing limits.

Reply:
If you have limits in place and you clear them that means you set all those limits to `""`. The keys themselves however will still be present and so the existence check for the key itself will yield true and so an update will be performed.

Reply:
It does?

If I "lxc config edit" and remove the key, then it won't be in the dict anymore. The value would be "" but the exists value would be False.

Reply:
That's not how I read the `Update()` method in `lxd/types/devices.go` that spews out the devices to update.